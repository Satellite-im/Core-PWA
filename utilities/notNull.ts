export default function notNull<TValue>(value: TValue | null): value is TValue {
  return value !== null
}
