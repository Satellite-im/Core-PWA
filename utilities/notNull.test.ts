import notNull from '~/utilities/notNull'

describe('notNull utility', () => {
  test('Expect undefined to be true', () => {
    const result: boolean = notNull(undefined)
    expect(result).toBe(true)
  })

  test('Expect null to be false', () => {
    const result: boolean = notNull(null)
    expect(result).toBe(false)
  })
})
