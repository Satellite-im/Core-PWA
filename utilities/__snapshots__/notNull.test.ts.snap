// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`notNull.default 0 1`] = `true`;

exports[`notNull.default 1 1`] = `false`;
